### PR TITLE
Comparison and Item lookup improvements

### PR DESCRIPTION
**Pull request by cameronwhite08**
I have added the ability to optionally:

- Find a drawing item clicked by a region (circle).
- Associate an identifier to drawing items (most useful when associating multiple items to a particular shape)
- Iterate the drawing items in reverse when looking up item for click event (items drawn front-to-back)
- Compare Paths and Points for value equality

### CODE REVIEW

File: lib/src/main/java/io/fabianterhorst/isometric/Point.java [hunk 1]
```diff
@@ -120,11 +122,63 @@ public double depth() {
     /**
      * Distance between two points
      */
-    public double distance(Point p1, Point p2) {
+    public static double distance(Point p1, Point p2) {
         double dx = p2.x - p1.x;
         double dy = p2.y - p1.y;
         double dz = p2.z - p1.z;
 
         return Math.sqrt(dx * dx + dy * dy + dz * dz);
```
Comment (FabianTerhorst) [inline lib/src/main/java/io/fabianterhorst/isometric/Point.java]:
```java
Math.sqrt(Point.distance2(p1, p2));
```

**Comment by FabianTerhorst** [comment]
All changes apart from the identifier are looking good, but i do think the lib shouldn't handle the identifiers because its always possible to extend the path or the shape even when this could be a bit more problematically.

**Comment by cameronwhite08** [comment]
That's a good idea, I had not thought of doing it that way! I've pushed some commits using that.

**Comment by FabianTerhorst** [comment]
Would merge it, are you done?

**Comment by cameronwhite08** [comment]
Yes i am good

**Pull request merged**